src/unity/toolkits/supervised_learning/supervised_learning.cpp

```diff
@@ -519,6 +519,14 @@ gl_sarray supervised_learning_model_base::fast_predict(
   } else if (output_type == "probability_vector"){
     ret_type = flex_type_enum::VECTOR;
   } else {
+      if (output_type == "probability" && variant_get_value<size_t>(state.at("num_classes")) > 2) {
+        std::stringstream ss;
+        ss << "Output type '" << output_type
```

Comment:
We don't need to use `output_type` here, since you've already tested that it's equal to `"probability"`. As a result, you can avoid the use of `std::stringstream` entirely